sgqlc/codegen/operation.py

```diff
@@ -456,9 +458,9 @@ def validate_required_arguments(self, node):
             if a['type']['kind'] == 'NON_NULL' and a['defaultValue'] is None:
                 required.add(a['name'])
 
-        for _, var in node.arguments:
-            if var.name in required:
-                required.remove(var.name)
+        for name, _ in node.arguments:
```

Comment:
could you check if this works with variables? See this fix I had to do sometime ago in a similar code e4a22f4f9b7d4bdf953673d1263930106f4195f1

I'd suggest to keep the old behavior if it's a variable, then:  `name, maybe_var` and you check `hasattr(maybe_var, 'name')`  to use it overriding `name`